src/Angle.cc

```diff
@@ -0,0 +1,41 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// \file
+/// \author Serge Monkewitz
+/// \brief This file contains the Angle class implementation.
+
+#include "Angle.h"
+
+#include <cstdio>
+#include <ostream>
+
+namespace lsst {
+namespace sg {
+
+std::ostream & operator<<(std::ostream & os, Angle const & a) {
+    char buf[32];
+    std::snprintf(buf, sizeof(buf), "%.17g rad", a.radians());
+    return os << buf;
+}
```

Comment:
This may be old school and a little clunky, but I far prefer it to writing a bunch of code that saves the iostream state, sets precision, unsets the floating point format (to get the equivalent of %g), calls a stream operator, then restores state. If you feel very strongly about this, I will reconsider.